Mathlib/CategoryTheory/Sites/ConstantSheaf.lean

```diff
@@ -0,0 +1,58 @@
+/-
+Copyright (c) 2023 Dagur Asgeirsson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Dagur Asgeirsson
+-/
+import Mathlib.CategoryTheory.Sites.Sheafification
+
+/-!
+
+# The constant sheaf
+
+We define the constant sheaf functor (the sheafification of the constant presheaf)
+`constantSheaf : D ⥤ Sheaf J D` and prove that it is left adjoint to evaluation at a terminal
+object (see `constantSheafAdj`).
+-/
+
+universe w v u
+
+namespace CategoryTheory
+
+open Limits Opposite
+
+variable {C : Type u} [Category.{v} C] (J : GrothendieckTopology C)
+
+variable (D : Type w) [Category.{max v u} D]
+
+/-- The constant presheaf functor is left adjoint to evaluation at a terminal object. -/
+noncomputable def constantPresheafAdj {T : C} (hT : IsTerminal T) :
```

Comment:
```suggestion
variable (D : Type w) [Category.{max v u} D]

/-- The constant presheaf functor is left adjoint to evaluation at a terminal object. -/
noncomputable def constantPresheafAdj {T : C} (hT : IsTerminal T) :
```
For this definition, the universe assumption on `D` could be relaxed.

Reply:
I just removed the explicit universes, Lean figures them out in the correct generality